packages/context/src/__tests__/unit/binding.unit.ts

```diff
@@ -214,6 +214,45 @@ describe('Binding', () => {
     });
   });
 
+  describe('cache', () => {
+    it('clears cache if scope changes', () => {
+      let index = 0;
+      const indexBinding = ctx
+        .bind<number>('index')
+        .toDynamicValue(() => {
+          return index++;
+        })
+        .inScope(BindingScope.SINGLETON);
+      expect(ctx.getSync(indexBinding.key)).to.equal(0);
+      // Singleton
+      expect(ctx.getSync(indexBinding.key)).to.equal(0);
+      indexBinding.inScope(BindingScope.CONTEXT);
+      expect(ctx.getSync(indexBinding.key)).to.equal(1);
```

Comment:
I find these test little bit difficult to understand, because I have to map from `0`/`1` values to whether the value getter was called or was not.

Could you please try to rewrite these tests using a Sinon spy? A mock-up to illustrate what I mean:

```ts
  const spy = sinon.spy();

// ...
  .toDynamicValue(spy);

  ctx.getSync(indexBinding.key)
  sinon.assertCalled(spy);

  ctx.getSync(indexBinding.key)
  sinon.assertNotCalled(spy);

  indexBinding.inScope(BindingScope.CONTEXT);
  sinon.assertCalled(spy);
```

The same comment applies to the next test too.

Reply:
👍